Add scoring support to KQL

lukasolson opened the pull request:
## Summary

Closes https://github.com/elastic/kibana/issues/86365.

* Adds scoring support to KQL, by providing an additional configuration option, `filtersInMustClause` (which defaults to `false`) when calling `buildEsQuery` (which in turn invokes `fromKuery` with this option).
* Changes `SearchSource` to send `filtersInMustClause: true` in cases where `_score` is inside the `sort` clause. (Scoring is disabled by Elasticsearch otherwise.)

![image](https://user-images.githubusercontent.com/1178348/123842892-1efdb600-d8c6-11eb-963a-99d9409b43bb.png)

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the [cloud](https://github.com/elastic/cloud) and added to the [docker list](https://github.com/elastic/kibana/blob/c29adfef29e921cc447d2a5ed06ac2047ceab552/src/dev/build/tasks/os_packages/docker_generator/resources/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)


### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](https://github.com/elastic/kibana/blob/master/RISK_MATRIX.mdx) |


### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### Release note

It is now possible to sort by `_score` in Discover when using a KQL query.

### Plugin API changes

`buildEsQuery` (and in turn, `fromKuery`) now support an additional option, `filtersInMustClause`, which will generate KQL queries in the `must` clause rather than the `filter` clause, in order to support use cases such as sorting by `_score`.

`SearchSource` now automatically sets `filtersInMustClause` to `true` when `_score` is included in the `sort` clause.

elasticmachine commented:
Pinging @elastic/kibana-app-services (Team:AppServices)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135262/)
* Commit: ed72d914b5349397a851edbb85ad025770a6efbe
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103727/ed72d914b5349397a851edbb85ad025770a6efbe)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135262/execution/node/190/log)

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135262/","number":"135262","commit":"ed72d914b5349397a851edbb85ad025770a6efbe"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135410/)
* Commit: abd4e52f02ec35c5e2405eeb2c20c6c3b7f01d59
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103727/abd4e52f02ec35c5e2405eeb2c20c6c3b7f01d59)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135410/execution/node/190/log)

### History
* :broken_heart: [Build #135262](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135262/) failed ed72d914b5349397a851edbb85ad025770a6efbe

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135262/","number":"135262","commit":"ed72d914b5349397a851edbb85ad025770a6efbe"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135410/","number":"135410","commit":"abd4e52f02ec35c5e2405eeb2c20c6c3b7f01d59"}]}
PIPELINE-->

flash1293 commented on x-pack/plugins/lens/public/indexpattern_datasource/field_item.test.tsx:
```diff
@@ -227,31 +227,14 @@ describe('IndexPattern Field Item', () => {
+      Array [
+        "/api/lens/index_stats/1/field",
+        Object {
+          "body": "{\\"dslQuery\\":{\\"bool\\":{\\"must\\":[],\\"filter\\":[],\\"should\\":[],\\"must_not\\":[]}},\\"fromDate\\":\\"now-7d\\",\\"toDate\\":\\"now\\",\\"fieldName\\":\\"bytes\\"}",
```
Seems like something broke here - the inline snapshot doesn't include the `geo.src: US` match. Didn't run the code

timroes commented:
![](https://media0.giphy.com/media/3o7WIPep7wkCI2rJ04/giphy.gif)

flash1293 commented:
I think this is a good change, but just to make sure - is there a performance hit attached to this?

YulNaumenko reviewed (approved):
Alerting related changes LGTM.

wylieconlon commented on x-pack/plugins/lens/public/indexpattern_datasource/field_item.test.tsx:
```diff
@@ -227,31 +227,14 @@ describe('IndexPattern Field Item', () => {
-            should: [],
-            must_not: [],
-          },
+    expect(core.http.post.mock.calls[0]).toMatchInlineSnapshot(`
```
Snapshot tests are weaker than exact match tests, are you sure that changing this to a snapshot is testing the right thing?

FrankHassanabad replied:
I'm wondering the same thing.

I am not opposed to it from our side, but one big con we have seen within `security_solutions` is that people tend to update the jest snapshots very quickly as a knee jerk reaction and then they will put the burden on the code reviewer to find where and if a mistake is made.

Typically if they have to at least update the objects, then we at least force them to see the changes by updating them manually vs. automatically. After a few code reviews where there were bugs and no one noticed, the snapshots began down trending from our side.

It's faster (true) if this output generator changes again which is a pro though. I do admit that.

I can approve this as-is with the inline snapshots unless someone else is strongly apposed from `security_solutions` and chimes in.

wylieconlon reviewed (commented):
I haven't tested this but here are my comments based on the PR:

1. What @timroes originally asked for was that a KQL consumer would be able to choose to use `filter` or  `must` depending on their needs. Is there a reason this is always on?

2. I think this implementation contradicts the [KQL](https://www.elastic.co/guide/en/kibana/current/kuery-query.html) or Discover docs, please update the docs in the same PR when making feature changes.

3. You've changed many of the tests to snapshots, this looks like something that needs an explanation

4. It does look like there is a performance hit as @flash1293 was asking, since [filter clauses can be cached](https://www.elastic.co/guide/en/elasticsearch/reference/7.13/query-dsl-bool-query.html) but not `must` clauses.

jgowdyelastic reviewed (approved):
ML changes LGTM

FrankHassanabad commented on x-pack/plugins/security_solution/public/common/components/top_n/index.test.tsx:
```diff
@@ -251,8 +251,8 @@ describe('StatefulTopN', () => {
 
-      expect(props.combinedQueries).toEqual(
-        '{"bool":{"must":[],"filter":[{"bool":{"filter":[{"bool":{"should":[{"match_phrase":{"network.transport":"tcp"}}],"minimum_should_match":1}},{"bool":{"should":[{"exists":{"field":"host.name"}}],"minimum_should_match":1}}]}},{"match_phrase":{"source.port":{"query":"30045"}}}],"should":[],"must_not":[]}}'
+      expect(props.combinedQueries).toMatchInlineSnapshot(
```
nit: I think this would be better if it was something which matched against an object rather than against a string.

Something like: `expect(JSON.parse(props.combinedQueries)).toMatchInlineSnaphshot(...)` if that works out. Otherwise this looks very hard to read and when people run the update jest command they're not going to look to see if they are fixing anything or not. But rather just mindlessly run it and do a commit and then the reviewer will have a hard time to read it since the string isn't being indented.

FrankHassanabad commented on x-pack/plugins/ml/public/application/jobs/new_job/utils/new_job_utils.test.ts:
```diff
@@ -140,24 +140,50 @@ describe('createSearchItems', () => {
-              bool: {
-                must_not: {
-                  bool: { should: [{ match_phrase: { airline: 'AWE' } }], minimum_should_match: 1 },
+    expect(resp).toMatchInlineSnapshot(`
```
Are you changing these so that way it's just easier to jest update them in the future with the jest update command?

I think this is fine for these here and a good idea.

FrankHassanabad reviewed (approved):
LGTM,

I did this:
* Looked over the code.
* Added a few questions about the inline snapshots within the tests as we normally don't use them. But still LGTM as I will not hold this up over those questions.

I did not do this:
* I did not check out the PR
* I did not test run it.

lukasolson commented:
Sorry for the noise here! And thanks for the feedback @wylieconlon and @flash1293.

I've updated this PR to conditionally allow generating KQL queries in the `filter` or `must` clause, and updated `SearchSource` to automatically generate them in the `must` clause if `_score` is inside of the `sort` options. I'd love to hear your feedback on this new approach.

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137086/)
* Commit: 53e3f7a5e6727d0391d7780d95550728acdf99e4
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Check Doc API Changes](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137086/execution/node/425/log)

### History
* :broken_heart: [Build #135410](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135410/) failed abd4e52f02ec35c5e2405eeb2c20c6c3b7f01d59
* :broken_heart: [Build #135262](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135262/) failed ed72d914b5349397a851edbb85ad025770a6efbe

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135262/","number":"135262","commit":"ed72d914b5349397a851edbb85ad025770a6efbe"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135410/","number":"135410","commit":"abd4e52f02ec35c5e2405eeb2c20c6c3b7f01d59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137086/","number":"137086","commit":"53e3f7a5e6727d0391d7780d95550728acdf99e4"}]}
PIPELINE-->

lukasolson commented:
@FrankHassanabad, sorry, didn't mean to re-request review

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137090/)
* Commit: 4ddc889e0f29806810d703413d68ade73505d379
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103727/4ddc889e0f29806810d703413d68ade73505d379)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/211a4634d8250abcb530781ee3ea435d62d59462) | [after](https://github.com/elastic/kibana/commit/4ddc889e0f29806810d703413d68ade73505d379) | diff |
| --- | --- | --- | --- |
| `data` | 3255 | 3246 | -9 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/211a4634d8250abcb530781ee3ea435d62d59462) | [after](https://github.com/elastic/kibana/commit/4ddc889e0f29806810d703413d68ade73505d379) | diff |
| --- | --- | --- | --- |
| `data` | 840.1KB | 840.5KB | +360.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/211a4634d8250abcb530781ee3ea435d62d59462) | [after](https://github.com/elastic/kibana/commit/4ddc889e0f29806810d703413d68ade73505d379) | diff |
| --- | --- | --- | --- |
| `data` | 3821 | 3812 | -9 |


</details>

### History
* :broken_heart: [Build #137086](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137086/) failed 53e3f7a5e6727d0391d7780d95550728acdf99e4
* :broken_heart: [Build #135410](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135410/) failed abd4e52f02ec35c5e2405eeb2c20c6c3b7f01d59
* :broken_heart: [Build #135262](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135262/) failed ed72d914b5349397a851edbb85ad025770a6efbe

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135262/","number":"135262","commit":"ed72d914b5349397a851edbb85ad025770a6efbe"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135410/","number":"135410","commit":"abd4e52f02ec35c5e2405eeb2c20c6c3b7f01d59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137086/","number":"137086","commit":"53e3f7a5e6727d0391d7780d95550728acdf99e4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137090/","number":"137090","commit":"4ddc889e0f29806810d703413d68ade73505d379"}]}
PIPELINE-->

FrankHassanabad reviewed (approved):
LGTM,

I did this:
* Looked over the code.
* Added a few questions about the inline snapshots within the tests as we normally don't use them. But still LGTM as I will not hold this up over those questions.

I did not do this:
* I did not check out the PR
* I did not test run it.

wylieconlon reviewed (approved):
Tested and LGTM, the automatic switching works as expected.

lukasolson commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138325/)
* Commit: 8b09edb66b37eda559fa4c7fae44555025caf73a
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103727/8b09edb66b37eda559fa4c7fae44555025caf73a)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/bdf1069e564b8e1229632b6a01d6122d1b563e70) | [after](https://github.com/elastic/kibana/commit/8b09edb66b37eda559fa4c7fae44555025caf73a) | diff |
| --- | --- | --- | --- |
| `data` | 3255 | 3246 | -9 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/bdf1069e564b8e1229632b6a01d6122d1b563e70) | [after](https://github.com/elastic/kibana/commit/8b09edb66b37eda559fa4c7fae44555025caf73a) | diff |
| --- | --- | --- | --- |
| `data` | 840.0KB | 840.4KB | +360.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/bdf1069e564b8e1229632b6a01d6122d1b563e70) | [after](https://github.com/elastic/kibana/commit/8b09edb66b37eda559fa4c7fae44555025caf73a) | diff |
| --- | --- | --- | --- |
| `data` | 3821 | 3812 | -9 |


</details>

### History
* :green_heart: [Build #137090](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137090/) succeeded 4ddc889e0f29806810d703413d68ade73505d379
* :broken_heart: [Build #137086](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137086/) failed 53e3f7a5e6727d0391d7780d95550728acdf99e4
* :broken_heart: [Build #135410](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135410/) failed abd4e52f02ec35c5e2405eeb2c20c6c3b7f01d59
* :broken_heart: [Build #135262](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135262/) failed ed72d914b5349397a851edbb85ad025770a6efbe

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135262/","number":"135262","commit":"ed72d914b5349397a851edbb85ad025770a6efbe"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135410/","number":"135410","commit":"abd4e52f02ec35c5e2405eeb2c20c6c3b7f01d59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137086/","number":"137086","commit":"53e3f7a5e6727d0391d7780d95550728acdf99e4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137090/","number":"137090","commit":"4ddc889e0f29806810d703413d68ade73505d379"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138325/","number":"138325","commit":"8b09edb66b37eda559fa4c7fae44555025caf73a"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140208/)
* Commit: 6a4a3b78abcffa3714f73a9c794465dfaf8c856a
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103727/6a4a3b78abcffa3714f73a9c794465dfaf8c856a)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/35695d8bb507ce07d081b51ae5165e82ba4ebc46) | [after](https://github.com/elastic/kibana/commit/6a4a3b78abcffa3714f73a9c794465dfaf8c856a) | diff |
| --- | --- | --- | --- |
| `infra` | 1.7MB | 1.7MB | +115.0B |
| `maps` | 3.2MB | 3.2MB | +115.0B |
| `securitySolution` | 6.3MB | 6.3MB | +257.0B |
| total |  |  | +487.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/35695d8bb507ce07d081b51ae5165e82ba4ebc46) | [after](https://github.com/elastic/kibana/commit/6a4a3b78abcffa3714f73a9c794465dfaf8c856a) | diff |
| --- | --- | --- | --- |
| `data` | 837.7KB | 838.1KB | +442.0B |

### History
* :green_heart: [Build #138325](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138325/) succeeded 8b09edb66b37eda559fa4c7fae44555025caf73a
* :green_heart: [Build #137090](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137090/) succeeded 4ddc889e0f29806810d703413d68ade73505d379
* :broken_heart: [Build #137086](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137086/) failed 53e3f7a5e6727d0391d7780d95550728acdf99e4
* :broken_heart: [Build #135410](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135410/) failed abd4e52f02ec35c5e2405eeb2c20c6c3b7f01d59
* :broken_heart: [Build #135262](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135262/) failed ed72d914b5349397a851edbb85ad025770a6efbe

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135262/","number":"135262","commit":"ed72d914b5349397a851edbb85ad025770a6efbe"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135410/","number":"135410","commit":"abd4e52f02ec35c5e2405eeb2c20c6c3b7f01d59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137086/","number":"137086","commit":"53e3f7a5e6727d0391d7780d95550728acdf99e4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137090/","number":"137090","commit":"4ddc889e0f29806810d703413d68ade73505d379"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138325/","number":"138325","commit":"8b09edb66b37eda559fa4c7fae44555025caf73a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140208/","number":"140208","commit":"6a4a3b78abcffa3714f73a9c794465dfaf8c856a"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140836/)
* Commit: 7b96af7eb790d6f64920fec31c0d666c3bd41087
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103727/7b96af7eb790d6f64920fec31c0d666c3bd41087)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Public APIs missing comments
> Total count of every public API that lacks a comment. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats comments` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/d3606aee04b5f97550760633852f491d87d628b2) | [after](https://github.com/elastic/kibana/commit/7b96af7eb790d6f64920fec31c0d666c3bd41087) | diff |
| --- | --- | --- | --- |
| `actions` | - | 117 | +117 |
| `advancedSettings` | - | 22 | +22 |
| `alerting` | - | 234 | +234 |
| `apm` | - | 39 | +39 |
| `apmOss` | - | 4 | +4 |
| `banners` | - | 9 | +9 |
| `bfetch` | - | 62 | +62 |
| `canvas` | - | 5 | +5 |
| `cases` | - | 407 | +407 |
| `charts` | - | 159 | +159 |
| `cloud` | - | 21 | +21 |
| `core` | - | 1081 | +1081 |
| `dashboard` | - | 137 | +137 |
| `dashboardEnhanced` | - | 50 | +50 |
| `dashboardMode` | - | 11 | +11 |
| `data` | - | 3162 | +3162 |
| `dataEnhanced` | - | 16 | +16 |
| `dataVisualizer` | - | 104 | +104 |
| `devTools` | - | 8 | +8 |
| `discover` | - | 55 | +55 |
| `discoverEnhanced` | - | 37 | +37 |
| `embeddable` | - | 384 | +384 |
| `embeddableEnhanced` | - | 14 | +14 |
| `encryptedSavedObjects` | - | 28 | +28 |
| `enterpriseSearch` | - | 2 | +2 |
| `esUiShared` | - | 90 | +90 |
| `eventLog` | - | 70 | +70 |
| `expressionError` | - | 12 | +12 |
| `expressionImage` | - | 4 | +4 |
| `expressionRepeatImage` | - | 28 | +28 |
| `expressionRevealImage` | - | 4 | +4 |
| `expressions` | - | 1569 | +1569 |
| `expressionShape` | - | 90 | +90 |
| `features` | - | 97 | +97 |
| `fileUpload` | - | 128 | +128 |
| `fleet` | - | 1033 | +1033 |
| `globalSearch` | - | 14 | +14 |
| `home` | - | 70 | +70 |
| `indexLifecycleManagement` | - | 4 | +4 |
| `indexManagement` | - | 157 | +157 |
| `indexPatternFieldEditor` | - | 29 | +29 |
| `infra` | - | 22 | +22 |
| `inspector` | - | 78 | +78 |
| `kibanaLegacy` | - | 62 | +62 |
| `kibanaReact` | - | 230 | +230 |
| `kibanaUtils` | - | 359 | +359 |
| `lens` | - | 190 | +190 |
| `licenseApiGuard` | - | 8 | +8 |
| `licenseManagement` | - | 3 | +3 |
| `licensing` | - | 42 | +42 |
| `lists` | - | 143 | +143 |
| `management` | - | 40 | +40 |
| `maps` | - | 203 | +203 |
| `mapsEms` | - | 75 | +75 |
| `metricsEntities` | - | 6 | +6 |
| `ml` | - | 274 | +274 |
| `monitoring` | - | 10 | +10 |
| `navigation` | - | 31 | +31 |
| `newsfeed` | - | 17 | +17 |
| `observability` | - | 219 | +219 |
| `osquery` | - | 11 | +11 |
| `presentationUtil` | - | 136 | +136 |
| `remoteClusters` | - | 4 | +4 |
| `reporting` | - | 132 | +132 |
| `rollup` | - | 20 | +20 |
| `ruleRegistry` | - | 60 | +60 |
| `runtimeFields` | - | 19 | +19 |
| `savedObjects` | - | 199 | +199 |
| `savedObjectsManagement` | - | 85 | +85 |
| `savedObjectsTagging` | - | 50 | +50 |
| `savedObjectsTaggingOss` | - | 50 | +50 |
| `screenshotMode` | - | 17 | +17 |
| `security` | - | 51 | +51 |
| `securityOss` | - | 9 | +9 |
| `securitySolution` | - | 1245 | +1245 |
| `share` | - | 83 | +83 |
| `snapshotRestore` | - | 22 | +22 |
| `spacesOss` | - | 5 | +5 |
| `stackAlerts` | - | 4 | +4 |
| `taskManager` | - | 25 | +25 |
| `telemetryCollectionManager` | - | 29 | +29 |
| `telemetryCollectionXpack` | - | 1 | +1 |
| `telemetryManagementSection` | - | 13 | +13 |
| `timelines` | - | 763 | +763 |
| `triggersActionsUi` | - | 228 | +228 |
| `uiActions` | - | 88 | +88 |
| `uiActionsEnhanced` | - | 147 | +147 |
| `urlForwarding` | - | 15 | +15 |
| `usageCollection` | - | 16 | +16 |
| `visTypeTimeseries` | - | 10 | +10 |
| `visualizations` | - | 229 | +229 |
| total |  |  | +15345 |

#### Any counts in public APIs
> Total count of every any typed public API. Target amount is 0. Run `node scripts/build_api_docs --plugin [yourplugin] --stats any` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/d3606aee04b5f97550760633852f491d87d628b2) | [after](https://github.com/elastic/kibana/commit/7b96af7eb790d6f64920fec31c0d666c3bd41087) | diff |
| --- | --- | --- | --- |
| `bfetch` | - | 1 | +1 |
| `charts` | - | 2 | +2 |
| `core` | - | 148 | +148 |
| `dashboard` | - | 1 | +1 |
| `data` | - | 98 | +98 |
| `dataVisualizer` | - | 3 | +3 |
| `embeddable` | - | 4 | +4 |
| `esUiShared` | - | 4 | +4 |
| `expressions` | - | 58 | +58 |
| `fileUpload` | - | 4 | +4 |
| `fleet` | - | 15 | +15 |
| `indexManagement` | - | 12 | +12 |
| `indexPatternFieldEditor` | - | 1 | +1 |
| `inspector` | - | 6 | +6 |
| `kibanaLegacy` | - | 3 | +3 |
| `kibanaReact` | - | 5 | +5 |
| `kibanaUtils` | - | 3 | +3 |
| `maps` | - | 2 | +2 |
| `mapsEms` | - | 1 | +1 |
| `ml` | - | 10 | +10 |
| `presentationUtil` | - | 3 | +3 |
| `reporting` | - | 1 | +1 |
| `savedObjects` | - | 3 | +3 |
| `savedObjectsTaggingOss` | - | 3 | +3 |
| `securitySolution` | - | 8 | +8 |
| `share` | - | 1 | +1 |
| `snapshotRestore` | - | 1 | +1 |
| `timelines` | - | 6 | +6 |
| `triggersActionsUi` | - | 1 | +1 |
| `uiActionsEnhanced` | - | 2 | +2 |
| `visTypeTimeseries` | - | 1 | +1 |
| `visualizations` | - | 13 | +13 |
| total |  |  | +424 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/d3606aee04b5f97550760633852f491d87d628b2) | [after](https://github.com/elastic/kibana/commit/7b96af7eb790d6f64920fec31c0d666c3bd41087) | diff |
| --- | --- | --- | --- |
| `infra` | 1.7MB | 1.7MB | +115.0B |
| `maps` | 3.2MB | 3.2MB | +115.0B |
| `securitySolution` | 6.4MB | 6.4MB | +257.0B |
| total |  |  | +487.0B |

#### Public APIs missing exports
> Total count of every type that is part of your API that should be exported but is not. This will cause broken links in the API documentation system. Target amount is 0. `Run node scripts/build_api_docs --plugin [yourplugin] --stats exports` for more detailed information.


| id | [before](https://github.com/elastic/kibana/commit/d3606aee04b5f97550760633852f491d87d628b2) | [after](https://github.com/elastic/kibana/commit/7b96af7eb790d6f64920fec31c0d666c3bd41087) | diff |
| --- | --- | --- | --- |
| `actions` | - | 7 | +7 |
| `advancedSettings` | - | 1 | +1 |
| `alerting` | - | 16 | +16 |
| `apm` | - | 30 | +30 |
| `bfetch` | - | 2 | +2 |
| `canvas` | - | 3 | +3 |
| `cases` | - | 14 | +14 |
| `charts` | - | 1 | +1 |
| `core` | - | 31 | +31 |
| `dashboard` | - | 9 | +9 |
| `data` | - | 64 | +64 |
| `dataEnhanced` | - | 2 | +2 |
| `devTools` | - | 2 | +2 |
| `discover` | - | 6 | +6 |
| `discoverEnhanced` | - | 2 | +2 |
| `embeddable` | - | 3 | +3 |
| `encryptedSavedObjects` | - | 3 | +3 |
| `esUiShared` | - | 1 | +1 |
| `eventLog` | - | 4 | +4 |
| `expressionError` | - | 2 | +2 |
| `expressionImage` | - | 1 | +1 |
| `expressionRevealImage` | - | 1 | +1 |
| `expressions` | - | 5 | +5 |
| `features` | - | 2 | +2 |
| `fileUpload` | - | 1 | +1 |
| `fleet` | - | 8 | +8 |
| `globalSearch` | - | 5 | +5 |
| `home` | - | 5 | +5 |
| `indexManagement` | - | 3 | +3 |
| `indexPatternFieldEditor` | - | 4 | +4 |
| `infra` | - | 3 | +3 |
| `inspector` | - | 4 | +4 |
| `kibanaLegacy` | - | 1 | +1 |
| `kibanaReact` | - | 4 | +4 |
| `kibanaUtils` | - | 8 | +8 |
| `lens` | - | 22 | +22 |
| `licensing` | - | 8 | +8 |
| `lists` | - | 38 | +38 |
| `management` | - | 5 | +5 |
| `maps` | - | 11 | +11 |
| `metricsEntities` | - | 1 | +1 |
| `ml` | - | 33 | +33 |
| `monitoring` | - | 2 | +2 |
| `navigation` | - | 2 | +2 |
| `observability` | - | 10 | +10 |
| `presentationUtil` | - | 5 | +5 |
| `reporting` | - | 14 | +14 |
| `ruleRegistry` | - | 9 | +9 |
| `runtimeFields` | - | 2 | +2 |
| `savedObjects` | - | 5 | +5 |
| `screenshotMode` | - | 1 | +1 |
| `security` | - | 6 | +6 |
| `securityOss` | - | 3 | +3 |
| `securitySolution` | - | 28 | +28 |
| `share` | - | 8 | +8 |
| `snapshotRestore` | - | 1 | +1 |
| `taskManager` | - | 8 | +8 |
| `telemetryCollectionManager` | - | 4 | +4 |
| `timelines` | - | 25 | +25 |
| `triggersActionsUi` | - | 19 | +19 |
| `uiActions` | - | 11 | +11 |
| `uiActionsEnhanced` | - | 10 | +10 |
| `usageCollection` | - | 2 | +2 |
| `visTypeTimeseries` | - | 3 | +3 |
| `visualizations` | - | 12 | +12 |
| total |  |  | +566 |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/d3606aee04b5f97550760633852f491d87d628b2) | [after](https://github.com/elastic/kibana/commit/7b96af7eb790d6f64920fec31c0d666c3bd41087) | diff |
| --- | --- | --- | --- |
| `data` | 837.7KB | 838.2KB | +442.0B |

<details><summary>Unknown metric groups</summary>

#### API count
| id | [before](https://github.com/elastic/kibana/commit/d3606aee04b5f97550760633852f491d87d628b2) | [after](https://github.com/elastic/kibana/commit/7b96af7eb790d6f64920fec31c0d666c3bd41087) | diff |
| --- | --- | --- | --- |
| `actions` | - | 117 | +117 |
| `advancedSettings` | - | 23 | +23 |
| `alerting` | - | 242 | +242 |
| `apm` | - | 39 | +39 |
| `apmOss` | - | 4 | +4 |
| `banners` | - | 9 | +9 |
| `bfetch` | - | 73 | +73 |
| `canvas` | - | 6 | +6 |
| `cases` | - | 445 | +445 |
| `charts` | - | 190 | +190 |
| `cloud` | - | 21 | +21 |
| `core` | - | 2360 | +2360 |
| `dashboard` | - | 160 | +160 |
| `dashboardEnhanced` | - | 51 | +51 |
| `dashboardMode` | - | 11 | +11 |
| `data` | - | 3716 | +3716 |
| `dataEnhanced` | - | 16 | +16 |
| `dataVisualizer` | - | 104 | +104 |
| `devTools` | - | 10 | +10 |
| `discover` | - | 81 | +81 |
| `discoverEnhanced` | - | 39 | +39 |
| `embeddable` | - | 456 | +456 |
| `embeddableEnhanced` | - | 14 | +14 |
| `encryptedSavedObjects` | - | 30 | +30 |
| `enterpriseSearch` | - | 2 | +2 |
| `esUiShared` | - | 92 | +92 |
| `eventLog` | - | 70 | +70 |
| `expressionError` | - | 12 | +12 |
| `expressionImage` | - | 4 | +4 |
| `expressionRepeatImage` | - | 28 | +28 |
| `expressionRevealImage` | - | 4 | +4 |
| `expressions` | - | 2003 | +2003 |
| `expressionShape` | - | 90 | +90 |
| `features` | - | 215 | +215 |
| `fileUpload` | - | 128 | +128 |
| `fleet` | - | 1128 | +1128 |
| `globalSearch` | - | 68 | +68 |
| `home` | - | 94 | +94 |
| `indexLifecycleManagement` | - | 4 | +4 |
| `indexManagement` | - | 162 | +162 |
| `indexPatternFieldEditor` | - | 31 | +31 |
| `infra` | - | 25 | +25 |
| `inspector` | - | 101 | +101 |
| `kibanaLegacy` | - | 66 | +66 |
| `kibanaReact` | - | 260 | +260 |
| `kibanaUtils` | - | 551 | +551 |
| `lens` | - | 206 | +206 |
| `licenseApiGuard` | - | 8 | +8 |
| `licenseManagement` | - | 3 | +3 |
| `licensing` | - | 117 | +117 |
| `lists` | - | 150 | +150 |
| `management` | - | 40 | +40 |
| `maps` | - | 204 | +204 |
| `mapsEms` | - | 75 | +75 |
| `metricsEntities` | - | 9 | +9 |
| `ml` | - | 278 | +278 |
| `monitoring` | - | 10 | +10 |
| `navigation` | - | 31 | +31 |
| `newsfeed` | - | 17 | +17 |
| `observability` | - | 219 | +219 |
| `osquery` | - | 11 | +11 |
| `presentationUtil` | - | 163 | +163 |
| `remoteClusters` | - | 4 | +4 |
| `reporting` | - | 133 | +133 |
| `rollup` | - | 20 | +20 |
| `ruleRegistry` | - | 60 | +60 |
| `runtimeFields` | - | 24 | +24 |
| `savedObjects` | - | 213 | +213 |
| `savedObjectsManagement` | - | 96 | +96 |
| `savedObjectsTagging` | - | 54 | +54 |
| `savedObjectsTaggingOss` | - | 89 | +89 |
| `screenshotMode` | - | 22 | +22 |
| `security` | - | 112 | +112 |
| `securityOss` | - | 12 | +12 |
| `securitySolution` | - | 1296 | +1296 |
| `share` | - | 123 | +123 |
| `snapshotRestore` | - | 23 | +23 |
| `spaces` | - | 106 | +106 |
| `spacesOss` | - | 72 | +72 |
| `stackAlerts` | - | 4 | +4 |
| `taskManager` | - | 52 | +52 |
| `telemetry` | - | 42 | +42 |
| `telemetryCollectionManager` | - | 29 | +29 |
| `telemetryCollectionXpack` | - | 1 | +1 |
| `telemetryManagementSection` | - | 14 | +14 |
| `timelines` | - | 882 | +882 |
| `triggersActionsUi` | - | 237 | +237 |
| `uiActions` | - | 127 | +127 |
| `uiActionsEnhanced` | - | 205 | +205 |
| `urlForwarding` | - | 15 | +15 |
| `usageCollection` | - | 57 | +57 |
| `visTypeTimeseries` | - | 10 | +10 |
| `visualizations` | - | 247 | +247 |
| total |  |  | +19277 |

#### References to deprecated APIs
| id | [before](https://github.com/elastic/kibana/commit/d3606aee04b5f97550760633852f491d87d628b2) | [after](https://github.com/elastic/kibana/commit/7b96af7eb790d6f64920fec31c0d666c3bd41087) | diff |
| --- | --- | --- | --- |
| `actions` | - | 8 | +8 |
| `alerting` | - | 32 | +32 |
| `apm` | - | 7 | +7 |
| `canvas` | - | 53 | +53 |
| `cases` | - | 151 | +151 |
| `crossClusterReplication` | - | 2 | +2 |
| `dashboard` | - | 128 | +128 |
| `dashboardEnhanced` | - | 10 | +10 |
| `dataEnhanced` | - | 53 | +53 |
| `dataVisualizer` | - | 16 | +16 |
| `discover` | - | 102 | +102 |
| `discoverEnhanced` | - | 19 | +19 |
| `embeddable` | - | 2 | +2 |
| `encryptedSavedObjects` | - | 2 | +2 |
| `fleet` | - | 89 | +89 |
| `globalSearch` | - | 4 | +4 |
| `graph` | - | 2 | +2 |
| `indexLifecycleManagement` | - | 2 | +2 |
| `indexManagement` | - | 12 | +12 |
| `infra` | - | 292 | +292 |
| `lens` | - | 160 | +160 |
| `lists` | - | 103 | +103 |
| `maps` | - | 592 | +592 |
| `ml` | - | 140 | +140 |
| `observability` | - | 34 | +34 |
| `presentationUtil` | - | 2 | +2 |
| `savedObjects` | - | 6 | +6 |
| `savedObjectsManagement` | - | 18 | +18 |
| `savedObjectsTaggingOss` | - | 5 | +5 |
| `security` | - | 2 | +2 |
| `securitySolution` | - | 881 | +881 |
| `stackAlerts` | - | 104 | +104 |
| `timelines` | - | 76 | +76 |
| `transform` | - | 16 | +16 |
| `uptime` | - | 11 | +11 |
| `urlDrilldown` | - | 18 | +18 |
| `visTypeTimeseries` | - | 6 | +6 |
| `visualizations` | - | 32 | +32 |
| total |  |  | +3192 |


</details>

### History
* :green_heart: [Build #140208](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140208/) succeeded 6a4a3b78abcffa3714f73a9c794465dfaf8c856a
* :green_heart: [Build #138325](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138325/) succeeded 8b09edb66b37eda559fa4c7fae44555025caf73a
* :green_heart: [Build #137090](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137090/) succeeded 4ddc889e0f29806810d703413d68ade73505d379
* :broken_heart: [Build #137086](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137086/) failed 53e3f7a5e6727d0391d7780d95550728acdf99e4
* :broken_heart: [Build #135410](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135410/) failed abd4e52f02ec35c5e2405eeb2c20c6c3b7f01d59

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/135410/","number":"135410","commit":"abd4e52f02ec35c5e2405eeb2c20c6c3b7f01d59"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137086/","number":"137086","commit":"53e3f7a5e6727d0391d7780d95550728acdf99e4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137090/","number":"137090","commit":"4ddc889e0f29806810d703413d68ade73505d379"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138325/","number":"138325","commit":"8b09edb66b37eda559fa4c7fae44555025caf73a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140208/","number":"140208","commit":"6a4a3b78abcffa3714f73a9c794465dfaf8c856a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140836/","number":"140836","commit":"7b96af7eb790d6f64920fec31c0d666c3bd41087"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142512/)
* Commit: d553c7b90cf3fb112111f97316ea00bbe80fb453
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103727/d553c7b90cf3fb112111f97316ea00bbe80fb453)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/f0e9210302003877394fd8e8ec9be23bb0be31a0) | [after](https://github.com/elastic/kibana/commit/d553c7b90cf3fb112111f97316ea00bbe80fb453) | diff |
| --- | --- | --- | --- |
| `infra` | 1.7MB | 1.7MB | +272.0B |
| `maps` | 3.2MB | 3.2MB | +272.0B |
| `securitySolution` | 6.4MB | 6.4MB | +415.0B |
| total |  |  | +959.0B |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/f0e9210302003877394fd8e8ec9be23bb0be31a0) | [after](https://github.com/elastic/kibana/commit/d553c7b90cf3fb112111f97316ea00bbe80fb453) | diff |
| --- | --- | --- | --- |
| `data` | 854.7KB | 855.3KB | +600.0B |

### History
* :green_heart: [Build #140836](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140836/) succeeded 7b96af7eb790d6f64920fec31c0d666c3bd41087
* :green_heart: [Build #140208](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140208/) succeeded 6a4a3b78abcffa3714f73a9c794465dfaf8c856a
* :green_heart: [Build #138325](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138325/) succeeded 8b09edb66b37eda559fa4c7fae44555025caf73a
* :green_heart: [Build #137090](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137090/) succeeded 4ddc889e0f29806810d703413d68ade73505d379
* :broken_heart: [Build #137086](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137086/) failed 53e3f7a5e6727d0391d7780d95550728acdf99e4

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137086/","number":"137086","commit":"53e3f7a5e6727d0391d7780d95550728acdf99e4"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137090/","number":"137090","commit":"4ddc889e0f29806810d703413d68ade73505d379"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138325/","number":"138325","commit":"8b09edb66b37eda559fa4c7fae44555025caf73a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140208/","number":"140208","commit":"6a4a3b78abcffa3714f73a9c794465dfaf8c856a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140836/","number":"140836","commit":"7b96af7eb790d6f64920fec31c0d666c3bd41087"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142512/","number":"142512","commit":"d553c7b90cf3fb112111f97316ea00bbe80fb453"}]}
PIPELINE-->

lizozom commented on packages/kbn-es-query/src/kuery/types.ts:
```diff
@@ -32,3 +32,8 @@ export interface KueryParseOptions {
 
 export { nodeTypes } from './node_types';
+
+export interface KueryQueryOptions {
```
Mark with @public

lizozom reviewed (approved):
Didn't test but did pair code review.
LGTM with minor comments.

I would add a warning to the `_score` column: If it's visible and showing 0's, users should click on it to actually get the scores.

lukasolson commented:
@elastic/kibana-app Could I get your feedback on this most recent change? If you don't sort by `_score`, then it won't retrieve values from Elasticsearch. So I added a small warning message to let the user know:

![image](https://user-images.githubusercontent.com/1178348/128428682-42043e58-18dc-4113-9119-7de3cf951559.png)

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143407/)
* Commit: d80dec4eee4132f603f00410deaf35f31000d532
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103727/d80dec4eee4132f603f00410deaf35f31000d532)
* This comment will update when the build is complete

### Failed CI Steps
* [Execute kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143407/execution/node/584/log)
* [Execute kibana-ciGroup1](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143407/execution/node/759/log)

### History
* :green_heart: [Build #142512](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142512/) succeeded d553c7b90cf3fb112111f97316ea00bbe80fb453
* :green_heart: [Build #140836](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140836/) succeeded 7b96af7eb790d6f64920fec31c0d666c3bd41087
* :green_heart: [Build #140208](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140208/) succeeded 6a4a3b78abcffa3714f73a9c794465dfaf8c856a
* :green_heart: [Build #138325](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138325/) succeeded 8b09edb66b37eda559fa4c7fae44555025caf73a
* :green_heart: [Build #137090](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137090/) succeeded 4ddc889e0f29806810d703413d68ade73505d379

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/137090/","number":"137090","commit":"4ddc889e0f29806810d703413d68ade73505d379"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138325/","number":"138325","commit":"8b09edb66b37eda559fa4c7fae44555025caf73a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140208/","number":"140208","commit":"6a4a3b78abcffa3714f73a9c794465dfaf8c856a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140836/","number":"140836","commit":"7b96af7eb790d6f64920fec31c0d666c3bd41087"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142512/","number":"142512","commit":"d553c7b90cf3fb112111f97316ea00bbe80fb453"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143407/","number":"143407","commit":"d80dec4eee4132f603f00410deaf35f31000d532"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143416/)
* Commit: ca184ff2f9a4f478a74c09908f78c07a8e1b8483
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103727/ca184ff2f9a4f478a74c09908f78c07a8e1b8483)
* This comment will update when the build is complete

### Failed CI Steps
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143416/execution/node/190/log)

### History
* :broken_heart: [Build #143407](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143407/) failed d80dec4eee4132f603f00410deaf35f31000d532
* :green_heart: [Build #142512](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142512/) succeeded d553c7b90cf3fb112111f97316ea00bbe80fb453
* :green_heart: [Build #140836](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140836/) succeeded 7b96af7eb790d6f64920fec31c0d666c3bd41087
* :green_heart: [Build #140208](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140208/) succeeded 6a4a3b78abcffa3714f73a9c794465dfaf8c856a
* :green_heart: [Build #138325](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138325/) succeeded 8b09edb66b37eda559fa4c7fae44555025caf73a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/138325/","number":"138325","commit":"8b09edb66b37eda559fa4c7fae44555025caf73a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140208/","number":"140208","commit":"6a4a3b78abcffa3714f73a9c794465dfaf8c856a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140836/","number":"140836","commit":"7b96af7eb790d6f64920fec31c0d666c3bd41087"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142512/","number":"142512","commit":"d553c7b90cf3fb112111f97316ea00bbe80fb453"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143407/","number":"143407","commit":"d80dec4eee4132f603f00410deaf35f31000d532"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143416/","number":"143416","commit":"ca184ff2f9a4f478a74c09908f78c07a8e1b8483"}]}
PIPELINE-->

lukasolson commented:
@elasticmachine merge upstream

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143934/)
* Commit: 3e856800c04a373c943984e14e6aca82a65ff236
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103727/3e856800c04a373c943984e14e6aca82a65ff236)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/5d9b9a15d89b2403473b67e3b9150fe3edbf7e75) | [after](https://github.com/elastic/kibana/commit/3e856800c04a373c943984e14e6aca82a65ff236) | diff |
| --- | --- | --- | --- |
| `discover` | 421 | 422 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/5d9b9a15d89b2403473b67e3b9150fe3edbf7e75) | [after](https://github.com/elastic/kibana/commit/3e856800c04a373c943984e14e6aca82a65ff236) | diff |
| --- | --- | --- | --- |
| `canvas` | 1.6MB | 1.6MB | +272.0B |
| `discover` | 509.8KB | 510.4KB | +561.0B |
| `infra` | 1.7MB | 1.7MB | +272.0B |
| `maps` | 3.2MB | 3.2MB | +272.0B |
| `securitySolution` | 6.5MB | 6.5MB | +415.0B |
| total |  |  | +1.8KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/5d9b9a15d89b2403473b67e3b9150fe3edbf7e75) | [after](https://github.com/elastic/kibana/commit/3e856800c04a373c943984e14e6aca82a65ff236) | diff |
| --- | --- | --- | --- |
| `data` | 790.5KB | 791.1KB | +600.0B |

### History
* :broken_heart: [Build #143416](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143416/) failed ca184ff2f9a4f478a74c09908f78c07a8e1b8483
* :broken_heart: [Build #143407](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143407/) failed d80dec4eee4132f603f00410deaf35f31000d532
* :green_heart: [Build #142512](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142512/) succeeded d553c7b90cf3fb112111f97316ea00bbe80fb453
* :green_heart: [Build #140836](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140836/) succeeded 7b96af7eb790d6f64920fec31c0d666c3bd41087
* :green_heart: [Build #140208](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140208/) succeeded 6a4a3b78abcffa3714f73a9c794465dfaf8c856a

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140208/","number":"140208","commit":"6a4a3b78abcffa3714f73a9c794465dfaf8c856a"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140836/","number":"140836","commit":"7b96af7eb790d6f64920fec31c0d666c3bd41087"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142512/","number":"142512","commit":"d553c7b90cf3fb112111f97316ea00bbe80fb453"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143407/","number":"143407","commit":"d80dec4eee4132f603f00410deaf35f31000d532"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143416/","number":"143416","commit":"ca184ff2f9a4f478a74c09908f78c07a8e1b8483"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143934/","number":"143934","commit":"3e856800c04a373c943984e14e6aca82a65ff236"}]}
PIPELINE-->

kibanamachine commented:
## :hourglass_flowing_sand: Build in-progress, with failures
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/144165/)
* Commit: e372c3fd98c834990906c0b1bc1793f5f138bfcb
* Storybooks not built yet
* This comment will update when the build is complete

### Failed CI Steps
* [Build OSS Plugins](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/144165/execution/node/179/log)
* [Report failed tests, if necessary](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/144165/execution/node/197/log)
* [Jest Unit Tests](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/144165/execution/node/190/log)

### History
* :green_heart: [Build #143934](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143934/) succeeded 3e856800c04a373c943984e14e6aca82a65ff236
* :broken_heart: [Build #143416](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143416/) failed ca184ff2f9a4f478a74c09908f78c07a8e1b8483
* :broken_heart: [Build #143407](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143407/) failed d80dec4eee4132f603f00410deaf35f31000d532
* :green_heart: [Build #142512](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142512/) succeeded d553c7b90cf3fb112111f97316ea00bbe80fb453
* :green_heart: [Build #140836](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140836/) succeeded 7b96af7eb790d6f64920fec31c0d666c3bd41087

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/140836/","number":"140836","commit":"7b96af7eb790d6f64920fec31c0d666c3bd41087"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142512/","number":"142512","commit":"d553c7b90cf3fb112111f97316ea00bbe80fb453"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143407/","number":"143407","commit":"d80dec4eee4132f603f00410deaf35f31000d532"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143416/","number":"143416","commit":"ca184ff2f9a4f478a74c09908f78c07a8e1b8483"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143934/","number":"143934","commit":"3e856800c04a373c943984e14e6aca82a65ff236"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/144165/","number":"144165","commit":"e372c3fd98c834990906c0b1bc1793f5f138bfcb"}]}
PIPELINE-->

kibanamachine commented:
## :green_heart: Build Succeeded
* [continuous-integration/kibana-ci/pull-request](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/144631/)
* Commit: f3cd94d851eaa1bfe27f9c58fec47c509ac293a5
* [Storybooks Preview](https://ci-artifacts.kibana.dev/storybooks/pr-103727/f3cd94d851eaa1bfe27f9c58fec47c509ac293a5)

### Metrics [[docs](https://www.elastic.co/guide/en/kibana/master/ci-metrics.html)]
#### Module Count
> Fewer modules leads to a faster build time


| id | [before](https://github.com/elastic/kibana/commit/a4f261a18714118cdfb0262225959556c87bb754) | [after](https://github.com/elastic/kibana/commit/f3cd94d851eaa1bfe27f9c58fec47c509ac293a5) | diff |
| --- | --- | --- | --- |
| `discover` | 429 | 430 | +1 |

#### Async chunks
> Total size of all lazy-loaded chunks that will be downloaded as the user navigates the app


| id | [before](https://github.com/elastic/kibana/commit/a4f261a18714118cdfb0262225959556c87bb754) | [after](https://github.com/elastic/kibana/commit/f3cd94d851eaa1bfe27f9c58fec47c509ac293a5) | diff |
| --- | --- | --- | --- |
| `canvas` | 1.6MB | 1.6MB | +272.0B |
| `discover` | 553.7KB | 554.2KB | +561.0B |
| `infra` | 1.8MB | 1.8MB | +415.0B |
| `lens` | 1.7MB | 1.7MB | +272.0B |
| `maps` | 3.2MB | 3.2MB | +272.0B |
| `securitySolution` | 6.5MB | 6.5MB | +415.0B |
| total |  |  | +2.2KB |

#### Page load bundle
> Size of the bundles that are downloaded on every page load. Target size is below 100kb


| id | [before](https://github.com/elastic/kibana/commit/a4f261a18714118cdfb0262225959556c87bb754) | [after](https://github.com/elastic/kibana/commit/f3cd94d851eaa1bfe27f9c58fec47c509ac293a5) | diff |
| --- | --- | --- | --- |
| `data` | 790.4KB | 791.0KB | +600.0B |

### History
* :broken_heart: [Build #144165](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/144165/) failed e372c3fd98c834990906c0b1bc1793f5f138bfcb
* :green_heart: [Build #143934](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143934/) succeeded 3e856800c04a373c943984e14e6aca82a65ff236
* :broken_heart: [Build #143416](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143416/) failed ca184ff2f9a4f478a74c09908f78c07a8e1b8483
* :broken_heart: [Build #143407](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143407/) failed d80dec4eee4132f603f00410deaf35f31000d532
* :green_heart: [Build #142512](https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142512/) succeeded d553c7b90cf3fb112111f97316ea00bbe80fb453

To update your PR or re-run it, just comment with:
`@elasticmachine merge upstream`

cc @lukasolson

<!--PIPELINE
{"builds":[{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/142512/","number":"142512","commit":"d553c7b90cf3fb112111f97316ea00bbe80fb453"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143407/","number":"143407","commit":"d80dec4eee4132f603f00410deaf35f31000d532"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143416/","number":"143416","commit":"ca184ff2f9a4f478a74c09908f78c07a8e1b8483"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/143934/","number":"143934","commit":"3e856800c04a373c943984e14e6aca82a65ff236"},{"status":"FAILURE","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/144165/","number":"144165","commit":"e372c3fd98c834990906c0b1bc1793f5f138bfcb"},{"status":"SUCCESS","url":"https://kibana-ci.elastic.co/job/elastic+kibana+pipeline-pull-request/144631/","number":"144631","commit":"f3cd94d851eaa1bfe27f9c58fec47c509ac293a5"}]}
PIPELINE-->
